src/Apps/ArtQuiz/artQuizRoutes.tsx

```diff
@@ -38,18 +38,68 @@ export const artQuizRoutes: AppRouteConfig[] = [
     children: [
       {
         path: "welcome",
-        hideFooter: true,
         getComponent: () => ArtQuizWelcome,
+        hideFooter: true,
+        query: graphql`
+          query artQuizRoutes_WelcomeQuery {
+            me {
+              quiz {
+                completedAt
+              }
+            }
+          }
+        `,
+        render: ({ Component, props }) => {
+          if (!(Component && props)) {
+            return
+          }
+          const { me } = props as any
+
+          if (!me?.quiz) {
+            return
+          }
+          const { completedAt } = me.quiz
+
+          if (completedAt) {
+            throw new RedirectException("/art-quiz/results")
+          }
+
+          return <Component {...props} />
+        },
       },
       {
         path: "artworks",
-        hideFooter: true,
         getComponent: () => ArtQuizArtworks,
+        hideFooter: true,
+        onClientSideRender: () => {
+          ArtQuizArtworks.preload()
+        },
+        query: graphql`
+          query artQuizRoutes_ArtworksQuery {
+            me {
+              quiz {
+                ...ArtQuizArtworks_quiz
+              }
+            }
+          }
+        `,
+        render: ({ Component, props }) => {
+          if (!(Component && props)) {
+            return
+          }
+          const { me } = props as any
+
+          if (!me?.quiz) {
+            return
+          }
+
+          return <Component {...props} quiz={me.quiz} />
```

Comment:
`blocking`: We should just follow how we do this elsewhere and the route component should have a fragment on `Me` rather than us pulling it off in this render function.

Reply:
rabble rabble